packages/ui-kit/src/components/markdown.tsx

```diff
@@ -298,36 +300,64 @@ const TypographyContainer = styled.div`
   ${containerStyles};
 `;
 
+const CopyArea = styled.div`
+  cursor: pointer;
+`;
+
+const TooltipBodyComponent = styled.div`
+  background-color: ${colors.light.surfaceCodeCopy};
+  color: ${colors.light.textInverted};
+  font-weight: ${typography.fontWeights.regular};
+  border-radius: ${tokens.borderRadiusForTooltip};
+  font-size: ${typography.fontSizes.extraSmall};
+  padding: ${dimensions.spacings.xs} ${dimensions.spacings.s};
+`;
+
 /* eslint-disable react/display-name */
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-const withAnchorLink = (Component: React.ComponentType) => (props: any) => {
-  return (
-    <Component
-      {...props}
-      css={css`
-        display: flex;
-        align-items: baseline;
-        :hover {
-          [data-link-type='anchor-link'] {
-            svg {
-              * {
-                fill: ${colors.light.linkNavigation};
+const withCopyToClipboard =
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (Component: React.ComponentType) => (props: any) => {
+    const [isCopiedToClipboard, setIsCopiedToClipboard] = React.useState(false);
+    const handleCopyToClipboardClick = (event: React.SyntheticEvent) => {
+      event.preventDefault();
+      const sectionUrl = `${window.location.href.split('#')[0]}#${props.id}`;
+      copyToClipboard(sectionUrl);
+      setIsCopiedToClipboard(true);
+      setTimeout(() => setIsCopiedToClipboard(false), 1500);
```

Comment:
Nitpick: the timeout based implementation works but has am edge case when you keep the pointer hovering over the item longer - the text changes back then, which seems erratic. 

Did you try using the `onClose` callback of the tooltip UI-Kit component to reset the state?  It's not in the "knobs" UI of the UI-Kit storybook but in the readme. 

https://uikit.commercetools.com/?path=/story/components-tooltips--tooltip

Reply:
@nkuehn works like a charm! Thank you, I guess we could implement something similar for the `code-block` component which still uses the timeout based approach: [`packages/ui-kit/src/components/code-block.tsx](https://github.com/commercetools/commercetools-docs-kit/blob/65b2345e6004dc7b69dbe76e3ede794ca5dc0906/packages/ui-kit/src/components/code-block.tsx#L213)
I'll create a ticket for that